batch.go

```diff
@@ -0,0 +1,58 @@
+package schnorrkel
+
+import (
+	"errors"
+
+	"github.com/gtank/merlin"
+	r255 "github.com/gtank/ristretto255"
+)
+
+// VerifyBatch batch verifies the given signatures
+func VerifyBatch(transcripts []*merlin.Transcript, signatures []*Signature, pubkeys []*PublicKey) (bool, error) {
```

Comment:
nit: this API could be error-prone. Since the user cant tell which key failed if they accidentally put a message out of order, the verification will fail and they will need to single verify everything again. 

an API similar to https://github.com/hdevalence/ed25519consensus/blob/main/batch.go helps prevent this type of thing. It could, also, be left up to the user to implement the above API to try to prevent the mistake.

Reply:
added a `BatchVerifier` that implements a similar API to what you linked